### ISSUE TITLE
API is not working

### ISSUE THREAD

**Issue by account0123** [account0123/RAE-API#2]
HTTP request returns 403

**Comment by Fontoj** [comment]
Came to ask something similar:

It seems that there were some changes in DLE's API and, at least in my testing some days ago, the first sign of broken token were in WOTD. since accessing to https://dle.rae.es/?m=wotd returns error.

Looking at the page, with today's date, the calling is as follows:

```
<div>
<h3 class="palabraT">Palabra del día</h3>
<p id="wotd" class="words"><a data-cat='WOTD' data-acc='WOTD' data-eti='amelcochar' href="/amelcochar?m=wotd2">amelcochar</a></p>
<p class="palabraD">domingo,  8 de agosto de 2021</p>
</div>

**Comment by victor141516** [comment]
I spent this afternoon investigating this thing:
Using curl I can call the API fine:
```sh
❯ curl -s 'https://dle.rae.es/data/search?w=hola'\
  -H 'authorization: Basic cDY4MkpnaFMzOmFHZlVkQ2lFNDM0' \
  -H 'user-agent: The app ;)'
{"approx":0,"res":[{"header":"hola","id":"KYtLWBc","grp":0}]}
```

But Node has a different opinion:
```js
import fetch from "node-fetch";

fetch('https://dle.rae.es/data/search?w=hola', {
    'headers': {
        'Authorization': 'Basic cDY4MkpnaFMzOmFHZlVkQ2lFNDM0',
        'User-Agent': 'The app ;)',
    },
}).then(r => r.text()).then(r => console.log(r))
```
⬆️⬆️ That prints a bunch of HTML and 403 as status

So whats the difference? I tried using [mitmproxy](https://github.com/mitmproxy/mitmproxy) to find out and:
Using this request I had HTTP 200:
```
Headers
Host:           dle.rae.es
Accept:         */*
authorization:  Basic cDY4MkpnaFMzOmFHZlVkQ2lFNDM0
user-agent:     The app ;)

Query
w: hola
```

But using this one I had HTTP 403:
```
Headers
Host:           dle.rae.es
Accept:         */*
authorization:  Basic cDY4MkpnaFMzOmFHZlVkQ2lFNDM0
user-agent:     The app ;)

Query
w: hola
```

Yeah they are exactly the same thing, so what the actual fuck?
At first I though it was because curl uses HTTP2 and node uses HTTP1.1 but I tried adding `--http1.1` to the curl args and it works fine too, so let's stick to 1.1 for now.\
\
Then after looking more closely I saw something called [ALPN](https://www.wikiwand.com/en/Application-Layer_Protocol_Negotiation) (I'll let you read that).
It turns out that the difference between those 2 requests is that curl makes the ALPN stuff by default. This is what mitmproxy shows:
```
Server Connection:
    Address           dle.rae.es:443
    Resolved Address  172.67.69.61:443
    HTTP Version      HTTP/1.1
    ALPN              http/1.1
```

But node does this:
```
Server Connection:
    Address           dle.rae.es:443
    Resolved Address  172.67.69.61:443
    HTTP Version      HTTP/1.1
```

Ok now comes the weird part. curl also has a `--no-alpn` arg so lets try that:
```sh
❯ curl -s --http1.1 --no-alpn -k \
  -H 'Host: dle.rae.es' \
  -H 'authorization: Basic cDY4MkpnaFMzOmFHZlVkQ2lFNDM0' \
  -H 'user-agent: The app ;)' \
  'https://localhost:8080/data/search?w=hola'
<shitload of HTML and 403>
```

NICE! We replicated the issue (!) ... kinda. That curl is going through mitmproxy so let's do the same but without it:
```sh
❯ curl -s --http1.1 --no-alpn \
  -H 'authorization: Basic cDY4MkpnaFMzOmFHZlVkQ2lFNDM0' \
  -H 'user-agent: The app ;)' \
  'https://dle.rae.es/data/search?w=hola'
{"approx":0,"res":[{"header":"hola","id":"KYtLWBc","grp":0}]}
```
What the... why this works? mitmproxy is not supposed to make any change to the request so what's happening here? I don't know, at this point I just began writting this comment and I expect someone more clever that me continues from here and solves the puzzle
\
Cheers

**Comment by account0123** [comment]
The API was updated, then i guess i should close this issue
